frontend/src/components/Schedule/AddYearButton.tsx

```diff
@@ -0,0 +1,91 @@
+import React from "react";
+import { connect } from "react-redux";
+import { Dispatch } from "redux";
+import { IPlanData } from "../../models/types";
+import {
+  getActivePlanFromState,
+  getIsUpdatingFromState,
+  getUserIdFromState,
+} from "../../state";
+import {
+  updateActivePlanAction,
+  updateActivePlanFetchingAction,
+} from "../../state/actions/userPlansActions";
+import { AppState } from "../../state/reducers/state";
+import { WhiteColorButton } from "../common/ColoredButtons";
+import { addYear } from "../../services/PlanService";
+import { convertPlanToUpdatePlanData } from "../../utils/plan-helpers";
+import { LoadingSpinner } from "../common/LoadingSpinner";
+
+type Props = ReduxStoreAddYearButtonProps & ReduxDispatchAddYearButtonProps;
+
+interface ReduxStoreAddYearButtonProps {
+  activePlan: IPlanData;
+  userId: number;
+  isUpdating: boolean;
+}
+
+interface ReduxDispatchAddYearButtonProps {
+  updateActivePlan: (newPlan: Partial<IPlanData>) => void;
+  updateActivePlanFetching: () => void;
+}
+
+const AddYearButtonComponent = ({
+  activePlan,
+  userId,
+  isUpdating,
+  updateActivePlan,
+  updateActivePlanFetching,
+}: Props) => {
+  const handleNewSemester = async () => {
+    updateActivePlanFetching();
+    const res = await addYear(
+      userId,
+      activePlan.id,
+      convertPlanToUpdatePlanData(activePlan, userId)
+    );
+
+    // Update store
+    updateActivePlan(res.plan);
+  };
+
+  return (
+    <WhiteColorButton
+      variant="contained"
+      style={{
+        width: "100%",
+        height: "36px",
+        border: "1px solid #eb5757",
+        marginBottom: "28px",
+        boxShadow: "0 0 1px rgb(255 255 255 / 50%)",
+        borderRadius: "0",
+      }}
+      disabled={isUpdating}
+      onClick={handleNewSemester}
+    >
+      + Add Year
+    </WhiteColorButton>
+  );
+};
+
+const mapStateToProps = (state: AppState) => ({
```

Comment:
Rather than use mapStateToProps and mapDispatchToProps, use the redux hooks api. See here for [reference](https://github.com/sandboxnu/graduatenu/blob/main/frontend/src/Onboarding/TransferCoursesScreen.tsx).

It's easier to read and also is more similar to the react hooks api, and [is the recommended api to use by redux docs](https://react-redux.js.org/api/hooks)-see tip at top of page.

Reply:
I think we have been using `mapStateToProps` and `mapDispatchToProps`. If we want, we should prob start using those hooks from now on 8)

Reply:
Yeet sounds good, i want

Reply:
LMAOOO "i want"

Reply:
`y e s`